### ISSUE TITLE
Can't use references

### ISSUE THREAD

**Issue by PrzemyslawKlys** [PowerShell/vscode-powershell#1696]
### System Details

System Details Output
```
### VSCode version: 1.30.2 61122f88f0bf01e2ac16bdb9e1bc4571755f5bd8 x64

### VSCode extensions:
aaron-bond.better-comments@2.0.3
abusaidm.html-snippets@0.2.1
adelphes.google-search-ext@0.2.1
AlanWalk.markdown-toc@1.5.6
alefragnani.Bookmarks@10.1.0
andrew-lis.workbench@0.1.7
azemoh.one-monokai@0.3.7
ban.spellright@3.0.22
batisteo.vscode-django@0.17.0
bierner.markdown-preview-github-styles@0.1.4
DavidAnson.vscode-markdownlint@0.23.0
DavidWang.ini-for-vscode@0.0.4
donjayamanne.githistory@0.4.4
donjayamanne.jupyter@1.1.4
donjayamanne.python-extension-pack@1.4.0
DotJoshJohnson.xml@2.4.0
dunstontc.dark-plus-syntax@0.0.149
eamodio.gitlens@9.4.1
ecmel.vscode-html-css@0.2.0
EditorConfig.EditorConfig@0.12.6
emilast.LogFileHighlighter@2.4.3
emmanuelbeziat.vscode-great-icons@2.1.46
Equinusocio.vsc-material-theme@2.6.3
esbenp.prettier-vscode@1.7.3
felixfbecker.php-debug@1.12.6
formulahendry.code-runner@0.9.5
gerane.Theme-dimmed-monokai@0.0.2
gharveymn.nightswitch@1.1.0
GlenBuktenica.unicode-substitutions@2.0.0
GrapeCity.gc-excelviewer@2.1.26
hdg.live-html-previewer@0.3.0
hnw.vscode-auto-open-markdown-preview@0.0.4
HookyQR.beautify@1.4.7
IBM.output-colorizer@0.1.2
ikappas.phpcs@1.0.5
ionutvmi.reg@0.0.2
JerryHong.autofilename@0.1.3
joaompinto.asciidoctor-vscode@2.3.0
johnpapa.winteriscoming@0.8.8
Kelvin.vscode-sshfs@1.11.2
kisstkondoros.vscode-codemetrics@1.17.4
KnisterPeter.vscode-github@0.30.1
liximomo.remotefs@0.0.13
magicstack.MagicPython@1.1.0
mariorodeghiero.vue-theme@1.1.2
mechatroner.rainbow-csv@0.8.0
mkloubert.vs-deploy@14.0.0
mkloubert.vscode-remote-workspace@0.38.0
ms-mssql.mssql@1.4.0
ms-python.python@2018.12.1
ms-vscode.csharp@1.17.1
ms-vscode.PowerShell@1.10.2
ms-vsliveshare.vsliveshare@0.3.1100
ms-vsts.team@1.144.1
msjsdiag.debugger-for-chrome@4.11.1
quicktype.quicktype@12.0.46
redhat.java@0.36.0
redhat.vscode-yaml@0.2.1
robertohuertasm.vscode-icons@8.0.0
seedess.vscode-remote-editor@0.0.1
Shan.code-settings-sync@3.2.4
stkb.rewrap@1.9.1
streetsidesoftware.code-spell-checker@1.6.10
streetsidesoftware.code-spell-checker-polish@1.0.0
swyphcosmo.spellchecker@1.2.13
techer.open-in-browser@2.0.0
TomasHubelbauer.zip-file-system@1.0.0
vangware.dark-plus-material@1.4.6
VisualStudioExptTeam.vscodeintellicode@1.1.2
vscjava.vscode-java-debug@0.16.0
vscjava.vscode-java-dependency@0.3.0
vscjava.vscode-java-pack@0.5.0
vscjava.vscode-java-test@0.13.0
vscjava.vscode-maven@0.13.0
WakaTime.vscode-wakatime@1.2.5
wayou.vscode-todo-highlight@1.0.4
whizkydee.material-palenight-theme@1.9.4
wholroyd.jinja@0.0.8
xaver.clang-format@1.8.0
zhuangtongfa.Material-theme@2.18.1
Zignd.html-css-class-completion@1.11.0


### PSES version: 1.10.2.0

### PowerShell version:

Name                           Value
----                           -----
PSVersion                      5.1.17763.134
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.134
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1


```

### Issue Description

I am experiencing a problem with...using references

![references problem](https://user-images.githubusercontent.com/15063294/51079409-f74f0f00-16c6-11e9-852c-820b47252d7a.gif)

[1547331489-3d373f1d-851d-41d4-840f-efffcfcf86231547331487236.zip](https://github.com/PowerShell/vscode-powershell/files/2752626/1547331489-3d373f1d-851d-41d4-840f-efffcfcf86231547331487236.zip)

I am not sure when this started...

**Comment by SydneyhSmith** [comment]
Looking through your logs we found: 
```
2019-01-12 23:59:07.142 [VERBOSE] tid:20 in 'ResolveFilePath' C:\PowerShellEditorServices\src\PowerShellEditorServices\Workspace\Workspace.cs: line 471
    Resolved path: c:\Support\GitHub\PSWriteHTML\Private\Get-HTMLLogos.ps1

2019-01-12 23:59:07.142 [VERBOSE] tid:20 in 'InvokeProviders' C:\PowerShellEditorServices\src\PowerShellEditorServices\Components\FeatureComponentBase.cs: line 71
    Invocation of provider 'Microsoft.PowerShell.EditorServices.Symbols.ScriptDocumentSymbolProvider' completed in 0ms.

2019-01-12 23:59:07.143 [VERBOSE] tid:20 in 'InvokeProviders' C:\PowerShellEditorServices\src\PowerShellEditorServices\Components\FeatureComponentBase.cs: line 71
    Invocation of provider 'Microsoft.PowerShell.EditorServices.Symbols.PsdDocumentSymbolProvider' completed in 0ms.

2019-01-12 23:59:07.143 [VERBOSE] tid:20 in 'InvokeProviders' C:\PowerShellEditorServices\src\PowerShellEditorServices\Components\FeatureComponentBase.cs: line 71
    Invocation of provider 'Microsoft.PowerShell.EditorServices.Symbols.PesterDocumentSymbolProvider' completed in 0ms.

2019-01-12 23:59:07.143 [VERBOSE] tid:20 in 'WriteMessage' C:\PowerShellEditorServices\src\PowerShellEditorServices.Protocol\MessageProtocol\MessageWriter.cs: line 88
    Writing Response 'textDocument/documentSymbol' with id 1113
    
    {
      "jsonrpc": "2.0",
      "id": "1113",
      "result": [
        {
          "name": "Get-HTMLLogos { }",
          "kind": 12,
          "location": {
            "uri": "file:///",
            "range": {
              "start": {
                "line": 0,
                "character": 0
              },
              "end": {
                "line": 43,
                "character": 1
              }
            }
          },
          "containerName": "Get-HTMLLogos"
        }
      ]
    }

2019-01-12 23:59:07.333 [VERBOSE] tid:20 in 'ReadMessage' C:\PowerShellEditorServices\src\PowerShellEditorServices.Protocol\MessageProtocol\MessageReader.cs: line 138
    Received Request 'textDocument/foldingRange' with id 1114
    
    {
      "jsonrpc": "2.0",
      "id": 1114,
      "method": "textDocument/foldingRange",
      "params": {
        "textDocument": {
          "uri": "file:///c%3A/Support/GitHub/PSWriteHTML/Private/Get-HTMLLogos.ps1"
        }
      }
    }

2019-01-12 23:59:07.333 [VERBOSE] tid:20 in 'ResolveFilePath' C:\PowerShellEditorServices\src\PowerShellEditorServices\Workspace\Workspace.cs: line 471
    Resolved path: c:\Support\GitHub\PSWriteHTML\Private\Get-HTMLLogos.ps1
```
it seems like the problem may be coming from the line where     `"uri": "file:///",` we will have to investigate more

**Comment by PrzemyslawKlys** [comment]
I believe this can be related to https://github.com/PowerShell/vscode-powershell/issues/1593 
When folder is within folder and then there are files it's more likely to happen.

![image](https://user-images.githubusercontent.com/15063294/51213580-a8e67e00-191b-11e9-88b0-75678755f2ea.png)

**Comment by SydneyhSmith** [comment]
@PrzemyslawKlys that issue does seem related, thanks for re-surfacing

**Comment by rkeithhill** [comment]
OK, as I suspected, this is the problem (I believe):
```
        "command": {
          "title": "6 references",
          "command": "editor.action.showReferences",
          "arguments": [
            "file:///c%3A/Support/GitHub/PSWriteHTML/Public/New-HTML.ps1",
            {
              "line": 0,
              "character": 0
            },
            [
              {
                "uri": "file:///c:/Support/GitHub/PSWriteHTML/Examples/Example1/PSHTML-AD.ps1",
                "range": {
                  "start": {
                    "line": 1509,
                    "character": 15
                  },
                  "end": {
                    "line": 1509,
                    "character": 23
                  }
                }
              },
              {
                "uri": "file:///c:/Support/GitHub/PSWriteHTML/Examples/Example2/Example2.ps1",
                "range": {
                  "start": {
                    "line": 13,
                    "character": 10
                  },
                  "end": {
                    "line": 13,
                    "character": 18
                  }
                }
              },
```
Notice the Uri `file:///c:` is incorrect.  It need to be `file:///c%3A`.  I think there are some other rules to this specific LSP URI format that @SeeminglyScience has coded around.  Basically this happens when PSES finds the filenames (instead of being given the URI from VSCode).  Our ScriptFile.ClientPath property is not returning a correct URI.  We've seen this issue with PSSA code analysis results.  The file listed in the problems pane can't be opened by double-clicking on them because of this issue.

**Comment by rkeithhill** [comment]
A fix for this should be coming in the next update.  https://github.com/PowerShell/PowerShellEditorServices/pull/872

**Comment by PrzemyslawKlys** [comment]
Yupi, is it addressing all other referenced issues?

**Comment by rkeithhill** [comment]
I believe this particular issue has been addressed by 1.12.  If you find other problems, please open a new issue.
